GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,201 @@
+# Gamepad Button and Axis Events
+
+## Authors:
+
+- Sneha Agarwal (https://github.com/snehagarwal_microsoft)
+- Steve Becker (https://github.com/SteveBeckerMSFT)
+
+## Participate
+- [Issue tracker]
+- [Discussion forum]
+
+## Status of this Document
+
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to the problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+
+- This document status: **Active**
+- Expected venue:**[W3C Web Applications Working Group](https://www.w3.org/groups/wg/webapps/)**
+- Current version: **This document**
+
+## Introduction
+
+The current Gamepad API relies on continuous polling to detect input changes, which can lead to input latency and increased CPU usage. This proposal introduces a more efficient approach that combines event-driven input handling for button presses and releases with frame-based state consolidation, where a single event encapsulates all gamepad state changes that occur within an input frame.
+
+- Event-Driven Input Handling: Immediate events (buttondown, buttonup) fire when discrete inputs like button presses or releases occur.
+
+- Frame-Based Consolidation: gamepadchange event encapsulates all the gamepad state changes that occurred within an input frame, preventing event spam.
+
+This proposal builds upon earlier work by Chromium developers while refining the hybrid model for improved performance.
+
+## User-Facing Problem
+
+The Gamepad API does not natively support event-driven input handling, and applications must rely only on continuous polling for updates. This polling-based approach introduces input latency because scripts cannot sync their polling with the arrival of new input, leading to a delay between input detection and handling. If an application polls at a regular interval, the average added latency is half the polling interval. For example, polling at 60Hz results in a latency of about 8.33 milliseconds. While decreasing the polling interval can reduce this latency, it comes at the cost of increased CPU usage.
+
+### Goals
+
+Reduce input latency by moving away from constant polling and introducing event-driven input handling. 
+
+### Non-goals
+
+The existing polling mechanism will not be deprecated at all. We are just proposing an alternative way of handling input events and applications are free to select whichever they prefer.
+
+## User research
```

Comment:
I think we can cut the user research section.  Let's focus on our proposed implementation instead of Firefox's old prototype.